adaptive/learner/averagelearner1D.py

```diff
@@ -0,0 +1,399 @@
+
+    def _update_data(self, x, y, point_type):
+        if point_type == "new":
+            self.data[x] = y
+        elif point_type == "resampled":
+            n = len(self._data_samples[x])
+            new_average = self.data[x] * n / (n + 1) + y / (n + 1)
+            self.data[x] = new_average
+
+    def _update_data_structures(self, x, y, point_type):
+        if point_type == "new":
+            self._data_samples[x] = [y]
+
+            if not self.bounds[0] <= x <= self.bounds[1]:
+                return
+
+            self._update_neighbors(x, self.neighbors_combined)
+            self._update_neighbors(x, self.neighbors)
+            self._update_scale(x, y)
+            self._update_losses(x, real=True)
+
+            # If the scale has increased enough, recompute all losses.
+            if self._scale[1] > self._recompute_losses_factor * self._oldscale[1]:
+                for interval in reversed(self.losses):
+                    self._update_interpolated_loss_in_interval(*interval)
+                self._oldscale = deepcopy(self._scale)
+
+            self._number_samples[x] = 1
+            self._undersampled_points.add(x)
+            self._error_in_mean[x] = np.inf
+            self._rescaled_error_in_mean[x] = np.inf
+            self._update_distances(x)
+            self._update_rescaled_error_in_mean(x, "new")
+
+        elif point_type == "resampled":
+            self._data_samples[x].append(y)
+            ns = self._number_samples
+            ns[x] += 1
+            n = ns[x]
+            if (x in self._undersampled_points) and (n >= self.min_samples):
+                x_left, x_right = self.neighbors[x]
+                if x_left is not None and x_right is not None:
+                    nneighbor = (ns[x_left] + ns[x_right]) / 2
+                elif x_left is not None:
+                    nneighbor = ns[x_left]
+                elif x_right is not None:
+                    nneighbor = ns[x_right]
+                else:
+                    nneighbor = 0
+                if n > self.neighbor_sampling * nneighbor:
+                    self._undersampled_points.discard(x)
+
+            # We compute the error in the estimation of the mean as
+            # the std of the mean multiplied by a t-Student factor to ensure that
+            # the mean value lies within the correct interval of confidence
+            y_avg = self.data[x]
+            ys = self._data_samples[x]
+            self._error_in_mean[x] = self._calc_error_in_mean(ys, y_avg, n)
+            self._update_distances(x)
+            self._update_rescaled_error_in_mean(x, "resampled")
+
+            if self._error_in_mean[x] <= self.min_Delta_g or n >= self.max_samples:
+                self._rescaled_error_in_mean.pop(x, None)
+
+            # We also need to update scale and losses
+            super()._update_scale(x, y)
+            self._update_losses_resampling(x, real=True)  #  XXX: REVIEW
+
+            # XXX: Is the following necessary?
```

Comment:
@AlvaroGI, do you know how to check whether this is needed?

Reply:
It is, I just removed this comment.